src/language/concurrency.md

```diff
@@ -16,90 +19,99 @@ prevpage:
+first clause - `http.get`, and return a `Future`. It will also tell the event
+loop to take note that sometime in the future, a specific event will happen (
+which is in this case an HTTP response), and when that event happens, it should
+execute the callback passed to `then`.
 
-As another example, consider the dart:io methods for reading files.
-The synchronous `File` method [`readAsStringSync()`][]
-reads a file synchronously,
-blocking until the file is either fully read or an error occurs.
-The method then either returns an object of type `String`
-or throws an exception.
-The asynchronous equivalent, [`readAsString()`][],
-immediately returns an object of type `Future<String>`.
-At some point in the future,
-the `Future<String>` completes with either a string value or an error.
+![Figure showing async events being added to an event loop and holding onto 
+a callback to execute later.](/assets/img/language/concurrency/async-event-loop.png)
 
-[`readAsStringSync()`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-io/File/readAsStringSync.html
-[`readAsString()`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-io/File/readAsString.html
+This same model is basically how the event loop handles all other 
+asynchronous events in Dart, such as [`Streams`][].
+
+[`Streams`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-async/Stream-class.html
+
+## Asynchrony types and syntax
 
+If you're already familiar with `Future`, `Stream`, and async-await,
+then you can skip ahead to the [isolates section][].
 
-#### Why asynchronous code matters
+[isolates section]: #isolates
 
-It matters whether a method is synchronous or asynchronous
-because most apps need to do more than one thing at a time.
+### Future type syntax
 
-Asynchronous computations are often the result of performing computations
-outside of the current Dart code; 
-this includes computations that don't complete immediately, 
-and where you aren't willing to block your Dart code waiting for the result.
-For example, an app might start an HTTP request,
-but need to update its display or respond to user input
-before the HTTP request completes.
-Asynchronous code helps apps stay responsive.
+A promise to eventually provide a string value is typed as `Future<String>`.
+This is an example of a function that returns a Future:
 
-These scenarios include operating system calls like
-non-blocking I/O, performing an HTTP request, or communicating with a browser. 
-Other scenarios include waiting for computations
-performed in another Dart isolate as described below, 
-or maybe just waiting for a timer to trigger. 
-All of these processes either run in a different thread, 
-or are handled by the operating system or the Dart runtime, 
-which allows Dart code to run concurrently with the computation.
+<?code-excerpt "lib/future_syntax.dart"?>
+```dart
+Future<String> _readFileAsync(String filename) {
+  final file = File(filename);
 
+  // .then() returns a Future
```

Comment:
There's the answer to my earlier question. So I guess you were just introducing a very high-level idea before and this section goes more in depth. I think you could either put this detail in that earlier paragraph, or at least say in that paragraph that the exact mechanics of what you're describing will be covered later (I prefer the former)

Reply:
I'm not sure what you're referring to.

Reply:
I've also realized that this code comment is misleading. Both `readAsString` and `then` return Futures. `then` is a function on the Future class (that also returns a future.)  I updated that code comment, and I'll update the text if you can clarify it for me.

Reply:
I was referring to my questions here, which you addressed. So I'll resolve this